src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -542,19 +553,20 @@ public SAMRecord next() {
             return getNextRecord();
         }
 
-        private SAMRecord getNextRecord() {
+        protected SAMRecord getNextRecord() {
             final SAMRecord result = nextRec;
             nextRec = null;
-            while(nextRec == null && unfilteredIterator.hasNext()) {
+            while (nextRec == null && unfilteredIterator.hasNext()) {
                 SAMRecord nextRecord = unfilteredIterator.next();
-                switch(compareToFilter(nextRecord)) {
+                switch (compareToFilter(nextRecord)) {
                     case MATCHES_FILTER:
                         nextRec = nextRecord;
                         break;
                     case CONTINUE_ITERATION:
                         continue;
                     case STOP_ITERATION:
-                        break;
+                        nextRec = null;
```

Comment:
Under what circumstances can you reach this line and have `nextRec` not be null? It's not modified outside of this method. Making it a `private` field would make this a little clearer.

Reply:
Yes, the assignment seems to be redundant. fixed.